build_tools/aws-sdk-code-generator/spec/protocol_tests_spec_helper.rb

```diff
@@ -0,0 +1,291 @@
+# frozen_string_literal: true
+
+require 'rexml/document'
+require 'rspec/expectations'
+
+PROTOCOL_TESTS_IGNORE_LIST_PATH = File.join(File.dirname(__FILE__), 'protocol-tests-ignore-list.json')
+
+# This module contains helpers relating to protocol-tests
+module ProtocolTestsHelper
+  class << self
+
+    def check_ignore_list(protocol, test_id, test_type)
+      return nil if protocol.include?('extras')
+
+      ignore_list = JSON.parse(File.read(PROTOCOL_TESTS_IGNORE_LIST_PATH))[protocol]
+      ignore_list[test_type].find { |i| i.key?(test_id) }
+    end
+
+    # sets up paths for each protocol and its file paths
+    def fixtures
+      fixtures = {}
+      files = Dir.glob(File.join(File.dirname(__FILE__), 'protocol-tests', '**', '*.json'))
+      raise 'unable to locate spec files' if files.empty?
+
+      files.each do |path|
+        protocol = File.basename(path).split('.').first
+        input_output = path.split('/')[-2]
+        fixtures[protocol] ||= {}
+        fixtures[protocol][input_output] = path
+      end
+      fixtures
+    end
+
+    def each_test_case(context, fixture_path)
+      return unless fixture_path
+
+      Aws::Json.load_file(fixture_path).each do |suite|
```

Comment:
Minor. JSON.load is fine here - Aws::Json is specifically for engine usage in the SDK (and correctly used below). Reading fixtures, I'd use JSON

Reply:
Yeah, that totally makes sense - I think this section was something that I grandfathered in from old test runner. 

Although, I would lean toward leaving as it is - to keep everything on this file consistent?

Reply:
Technically it was wrong before. Regardless of engine, we'd want this file parsed the same way each time. In theory all engines should parse this the same :shrug:

Reply:
Yeah, I'd lean towards consistency in the code of using the engine interface and always use AWS::Json instead of any specific library, even though this is test code vs runtime.  

Although - now that I'm looking, our specs use a mix of both approaches already :-/
* JSON.load: https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sdk-core/spec/api_helper.rb#L11
* Aws::Json: https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sdk-core/spec/aws/ecs_credentials_spec.rb#L305